Pa3/Durchführung/Project/app/StateHandler.c

```diff
@@ -31,13 +31,190 @@
 /* PROTOTYPES *************************************************************************************/
 
 /* VARIABLES **************************************************************************************/
-static EventEnum gCurrentState;
+static EventEnum gCurrentEvent = NO_EVENT_HAS_HAPPEND; 
+static StateEnum gCurrentState = STATE_INIZALIZATION_MCU; //First state when the MCU is started
 static SoftTimer* gLapTimer;
+static bool gEntryDone = false;
 
 /* EXTERNAL FUNCTIONS *****************************************************************************/
 
 void StateHandler_process(void)
 {
+  switch(gCurrentState)
+  { 
+    case STATE_INIZALIZATION_MCU: 
+      InitializeMcu_InitializeAll(void); //entry
+      InitializeMcu_DisplayTeamName(void); //exit
+      gCurrentState = STATE_CALIBRATE_LINE_SENSORS;
+      break;
+
+    case STATE_CALIBRATE_LINE_SENSORS:
+      if(gEntryDone == false)
+      {
+        gEntryDone = true;
+        gCurrentEvent = CalibrateLineSensors_Initialize(void); //entry
+        if (NO_EVENT_HAS_HAPPEND != gCurrentEvent)
+        {
+          gCurrentState = STATE_ERROR_HANDLER;
+        }
+      }
+
+      gCurrentEvent = CalibrateLineSensors_CalibrateSensors(void); //do
+      switch (gCurrentEvent)
+      {
+        case CALIBRATION_FAILED:
+          gCurrentState = STATE_ERROR_HANDLER;
+          break;
+
+        case CALIBRATION_DONE:
+          gEntryDone = false;
+          gCurrentState = STATE_READY_TO_DRIVE;
+          break;
+
+        default:
+          break; 
+      }
+      break;
+
+    case STATE_READY_TO_DRIVE:         
+      gCurrentEvent = ReadyToDrive_CheckStateOfButtons(void); //do
+
+      switch(gCurrentEvent)
+      {
+        case CALIBRATION_BUTTON_HAS_BEEN_RELEASED:
+          gCurrentState = STATE_CALIBRATE_LINE_SENSORS;
+          break;
```

Comment:
Since you determine the state change at this point this could be a great place for entry/exit functions without needing a flag. No need to do it like that, just a suggestion. This is also just an example your state machine does not have a ReadyToDrive_exit or Calibrate_entry() function I think
```suggestion
        case CALIBRATION_BUTTON_HAS_BEEN_RELEASED:
          ReadyToDrive_exit();
          gCurrentState = STATE_CALIBRATE_LINE_SENSORS;
          Calibrate_entry()
          break;
```